extensions/amp-fx-scroll-sync/0.1/scroll-sync-sticky-top-effect.js

```diff
@@ -0,0 +1,122 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {ScrollSyncEffect} from './scroll-sync-effect';
+import {setStyles, getStyle} from '../../../src/style';
+import {viewportFor} from '../../../src/viewport';
+import {insertAfter} from '../../../src/dom';
+
+const ELEMENT_SHIM_TAG = 'i-amp-scroll-sync-shim';
+
+export class ScrollSyncStickyTopEffect extends ScrollSyncEffect {
+
+  constructor(element, unusedConfig) {
+    super(element);
+    this.scrollMax_ = null;
+    this.scrollMin_ = null;
+    this.layoutBox_ = null;
+    this.isDocked = false;
+    this.elementShim_ = null;
+
+    this.win = element.ownerDocument.defaultView;
+    this.viewport_ = viewportFor(this.win);
+
+    // TODO: This is wrong and we shouldn't be doing it. But because of collapsing
+    // margins we are getting a "false" offsetTop position measurement.
+    setStyles(this.element, {
+      'overflow': 'auto',
+    });
+  }
+
+  /** @override */
+  isDirectional() {
+    return false;
+  }
+
+  /** @override */
+  measure() {
+    if (!this.scrollMin_) {
+      this.scrollMin_ = this.element.offsetTop;
+    }
+    if (!this.scrollMax_) {
+      this.scrollMax_ = this.scrollMin_ + 1;
+    }
+
+    this.layoutBox_ = this.viewport_.getLayoutRect(this.element);
```

Comment:
`getIntersectionElementLayoutBox` is used on an AMP element, but in the `ScrollSyncStickyTopEffect` class, `this.element` can be any dom element such as a `<div>`. I thought we cannot use `getIntersectionElementLayoutBox` on a dom element.